sustainability_stock/tests/common.py

```diff
@@ -0,0 +1,142 @@
+
+        cls.product_to_sell = cls.env["product.product"].create(
+            {"name": "Product to Sell", "type": "product", "weight": 1.0}
+        )
+        cls.product_to_purchase = cls.env["product.product"].create(
+            {"name": "Product to Purchase", "type": "product", "weight": 2.0}
+        )
+
+        product_delivery = cls.env["product.product"].create(
+            {
+                "name": "Delivery Charges",
+                "type": "service",
+                "list_price": 40.0,
+                "categ_id": cls.env.ref("delivery.product_category_deliveries").id,
+            }
+        )
+        delivery_carrier = cls.env["delivery.carrier"].create(
+            {
+                "name": "Standard Delivery",
+                "fixed_price": 40,
+                "delivery_type": "fixed",
+                "product_id": product_delivery.id,
+            }
+        )
+
+        product_uom_unit = cls.env.ref("uom.product_uom_unit")
+        so = cls.env["sale.order"].create(
+            {
+                "partner_id": cls.customer.id,
+                "partner_invoice_id": cls.customer.id,
+                "partner_shipping_id": cls.customer.id,
+                "order_line": [
+                    Command.create(
+                        {
+                            "name": cls.product_to_sell.name,
+                            "product_id": cls.product_to_sell.id,
+                            "product_uom_qty": 10,
+                            "product_uom": product_uom_unit.id,
+                            "price_unit": 120.00,
+                        },
+                    )
+                ],
+            }
+        )
+        delivery_wizard = Form(
+            cls.env["choose.delivery.carrier"].with_context(
+                default_order_id=so.id, default_carrier_id=delivery_carrier.id
+            )
+        )
+        delivery_wizard.save().button_confirm()
+        so.action_confirm()
+        cls.outgoing_picking = so.picking_ids[0]
+        # FIX shipping_weight is a non stored computed field, we cannot do that
+        cls.outgoing_picking.shipping_weight = 10.0
+        cls.outgoing_picking.write({"state": "done"})
+
+        po = cls.env["purchase.order"].create(
+            {
+                "partner_id": cls.vendor.id,
+                "order_line": [
+                    Command.create(
+                        {
+                            "product_id": cls.product_to_purchase.id,
+                            "product_qty": 5.0,
+                            "price_unit": 100.0,
+                            "name": "Purchase of " + cls.product_to_purchase.name,
+                        },
+                    )
+                ],
+            }
+        )
+        po.button_confirm()
+        cls.incoming_picking = po.picking_ids[0]
+        # FIX shipping_weight is a non stored computed field, we cannot do that
+        cls.incoming_picking.shipping_weight = 10.0
```

Comment:
if you have the time to fix this thanks :)
Or I'll do it end of next week

Reply:
i did it